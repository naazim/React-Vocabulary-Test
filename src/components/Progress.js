import React from 'react';

const Progress = (props) => (
    <div className="progress">
        <div
            className="progress__percent"
            style={{ width: props.percentage + '%' }}
        ></div>
    </div>
);

export default Progress;