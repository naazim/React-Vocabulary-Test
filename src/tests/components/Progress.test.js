import React from 'react';
import { shallow } from 'enzyme';
import Progress from '../../components/Progress';

test('should display the recieved progress value', () => {
    const wrapper = shallow(<Progress percentage="50" />);
    expect(wrapper.find('.progress__percent').prop('style').width).toBe('50%');
    expect(wrapper).toMatchSnapshot();
})

test('should display no progress if recieved progress value is 0', () => {
    const wrapper = shallow(<Progress percentage="0" />);
    expect(wrapper.find('.progress__percent').prop('style').width).toBe('0%');
    expect(wrapper).toMatchSnapshot();
})