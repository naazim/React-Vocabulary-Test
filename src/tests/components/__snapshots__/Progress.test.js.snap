// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display no progress if recieved progress value is 0 1`] = `
<div
  className="progress"
>
  <div
    className="progress__percent"
    style={
      Object {
        "width": "0%",
      }
    }
  />
</div>
`;

exports[`should display the recieved progress value 1`] = `
<div
  className="progress"
>
  <div
    className="progress__percent"
    style={
      Object {
        "width": "50%",
      }
    }
  />
</div>
`;
